### PR TITLE
Fix Docs spacing/indentation and RFC3339 time

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?
- [x] This PR is not related to any Issue

This PR fixes mixed tabs and spaces in indention, as well as instances of using tabs in alignment within the documentation.

This PR also changes several places in the API documentation that referred to dates/times being formatted according to RFC3339 that, in fact, use the custom `lib/go-tc.TimeLayout` format ("2006-01-02 15:04:05-07"). These times are now referred to as being "ISO-like".

## Which Traffic Control components are affected by this PR?
- Documentation

## What is the best way to verify this PR?
Make sure documentation builds with no errors or warnings, ensure indentation and alignment are consistent and in accordance with [the relevant documentation guidelines](https://traffic-control-cdn.readthedocs.io/en/latest/development/documentation_guidelines.html#indentation). Ensure all date/times that are _not_ presented in RFC3339 format are not documented as though they were, and that those that _are_ are correctly documented as being RFC3339.

## If this is a bug fix, what versions of Traffic Control are affected?
- master
- 5.x probably
- 4.x probably

## The following criteria are ALL met by this PR
- [x] Tests are unnecessary for documentation changes
- [x] This PR includes documentation
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Review by zrhoffman — CHANGES_REQUESTED** [review]
"An ISO-like format" is vague and does not accomplish the purpose of documentation, in this case, which is to allow a user to anticipate the date format.

The whitespace updates all look fine, so if you want, if you remove the *RFC3339 time* commit from the PR and throw it in a separate one for more discussion, I'll gladly merge the whitespace fixes.

**Comment by ocket8888** [comment]
> "An ISO-like format" is vague and does not accomplish the purpose of documentation, in this case, which is to allow a user to anticipate the date format.

"an ISO-like format" is consistent with how that format is described throughout most of the documentation, though I agree it's overly vague.

> The whitespace updates all look fine, so if you want, if you remove the RFC3339 time commit from the PR and throw it in a separate one for more discussion, I'll gladly merge the whitespace fixes.

Both changes ought to be done, so I don't really care much about splitting them up, unless you're saying I should do that.

Here's my first thought on improving that: change "an ISO-like format" to a link to a label in the API index section that describes the custom format in more detail, and also probably links to the format's GoDoc on go.pkg.dev.

**Comment by zrhoffman** [comment]
> Both changes ought to be done, so I don't really care much about splitting them up, unless you're saying I should do that.

Oh, splitting would only be if you want to avoid unnecessary merge conflicts if something else is merged before the RCF3339 part is merged, keeping them together works for me.

**Comment by zrhoffman** [comment]
> Here's my first thought on improving that: change "an ISO-like format" to a link to a label in the API index section that describes the custom format in more detail, and also probably links to the format's GoDoc on go.pkg.dev.

The links sound good. I think providing a term to describe it would still be helpful, even if it means inventing one.

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -44,7 +44,7 @@ Glossary
...
+		There are two basic types of :dfn:`Cache Groups`: EDGE_LOC and MID_LOC ("LOC" being short for "location" - a holdover from when :dfn:`Cache Groups` were called "Cache Locations). Traffic Control is a two-tiered system, where the clients get directed to the Edge-tier (EDGE_LOC) :dfn:`Cache Group`. On cache miss, the :term:`cache server` in the Edge-tier :dfn:`Cache Group` obtains content from a Mid-tier (MID_LOC) :dfn:`Cache Group`, rather than the origin, which is shared with multiple Edge-tier :dfn:`Cache Groups`. Edge-tier :dfn:`Cache Groups` are usually configured to have a single "parent" :dfn:`Cache Group`, but in general Mid-tier :dfn:`Cache Groups` have many "children".
```
Comment (zrhoffman) [inline docs/source/glossary.rst:47]: > Traffic Control is a two-tiered system

With the introduction of Flexible Topologies, Traffic Control is no longer a strictly two-tiered system, but, since this PR relates to whitespace and the TO date format, rewording this should go in a separate PR.

File: docs/source/api/index.rst [hunk 2]
```diff
@@ -124,6 +124,14 @@ The following, reserved properties of ``summary`` are guaranteed to always have
...
 
+.. _non-rfc-datetime:
+
+Traffic Ops's Custom Date/Time Format
+-------------------------------------
+Traffic Ops will often return responses from its API that include dates. As of the time of this writing, the vast majority of those dates are written in a non-:rfc:3339 format (this is tracked by :issue:`5911`). This is most commonly the case in the ``last_updated`` properties of objects returned as JSON-encoded documents. The format used is :samp:`{YYYY}-{MM}-{DD} {hh}:{mm}:{ss}±{ZZ}`, where ``YYYY`` is the 4-digit year, ``MM`` is the two-digit (zero padded) month, ``DD`` is the two-digit (zero padded) day of the month, ``hh`` is the two-digit (zero padded) hour of the day, ``mm`` is the two-digit (zero padded) minute of the hour, ``ss`` is the two-digit (zero padded) second of the minute, and ``ZZ`` is the two-digit (zero padded) timezone offset in hours of the date/time's local timezone from UTC (the offset can be positive or negative as indicated by a ``+`` or a ``-`` directly before it, where the sample has a ``±``).
```
Comment (zrhoffman) [inline docs/source/api/index.rst:131]: Should this definition include an example?

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Review by zrhoffman — APPROVED** [review]
